### PR TITLE
Kernel v3 compatibility

### LINKED ISSUE #135: MatcherFactory Broken with ezplatform v3
MatcherFactory extends a `final` class and throws error

Introduced by updstream https://github.com/ezsystems/ezpublish-kernel/commit/177d87da3cec7dc2bf070c130ce95552479a9f04

### PR DESCRIPTION
**Pull request by pspanja**
Fixes #135 

Made compatible with v3 of the kernel:

- `MatcherFactory` depended on a class that was made final, see #135 
- `RepositoryFactory` got a new parameter

### CODE REVIEW

**Comment by pspanja** [comment]
Ping @wizhippo

File: bundle/Matcher/MatcherFactory.php [hunk 1]
```diff
@@ -32,13 +43,15 @@ class MatcherFactory extends ServiceAwareMatcherFactory
     private $scope;
 
     public function __construct(
+        ?ViewMatcherRegistry $viewMatcherRegistry,
```
Comment (emodric) [inline bundle/Matcher/MatcherFactory.php:46]: So, no issues if class does not exist (eZ Platform 2.5)?

File: bundle/Matcher/MatcherFactory.php [hunk 2]
```diff
@@ -54,4 +67,27 @@ public function match(View $view): ?array
...
+     *
+     * @return MatcherInterface|ViewMatcherInterface
+     */
+    protected function getMatcher($matcherIdentifier)
+    {
+        if ($this->viewMatcherRegistry !== null && \strpos($matcherIdentifier, '@') === 0) {
+            return $this->viewMatcherRegistry->getMatcher(\substr($matcherIdentifier, 1));
+        }
+
+        if ($this->container->has($matcherIdentifier)) {
```
Comment (emodric) [inline bundle/Matcher/MatcherFactory.php:84]: Do we need this still? This is for eZ Platform 2.5 support?

File: bundle/Matcher/MatcherFactory.php (hunk 1 above)
Comment (pspanja) [reply bundle/Matcher/MatcherFactory.php:46]: None, apparently in PHP this is safe.

File: bundle/Matcher/MatcherFactory.php (hunk 2 above)
Comment (pspanja) [reply bundle/Matcher/MatcherFactory.php:84]: Yes, in case someone references a matcher by service name that does not begin with `@`.

**Review by emodric — APPROVED** [review]

**Comment by wizhippo** [comment]
media-site/ezplatform3 is closer to working now with these changes.  Thanks.

**Comment by emodric** [comment]
@wizhippo What does not work with `media-site/ezplatform3` ? Works fine for me (after this patch) :)

**Comment by wizhippo** [comment]
I may have some dep conflicts.  SncRedisBundle is complaining about empty client config and building the ezplatform assets is complaining.  I've spun up a media site using the doc/docker.  Working now after a composer update.

**Comment by emodric** [comment]
> SncRedisBundle is complaining about empty client config

Hm... I thought I fixed that in the repo. Will check again.

> building the ezplatform assets is complaining

Complaining how?

**Comment by wizhippo** [comment]
> Complaining how?

It's all good now thanks

**Comment by pspanja** [comment]
Updated for https://github.com/ezsystems/ezpublish-kernel/pull/2808 in https://github.com/netgen/ezplatform-site-api/pull/137/commits/bf2ad5fd4fd51c047782601b3f2e9b8a193e33f4.

**Comment by pspanja** [comment]
@emodric @wizhippo thanks for the review!

**Pull request merged**